### PR TITLE
fix: sleeper behavior in data scanner

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: sleeper behavior in data scanner

## Motivation and Context
do not apply healReplication() for ILM
expired, transitioned objects

## How to test this PR?
Manually inspecting the code

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 12164-c005cde/gopath/src/github.com/minio
+ git -C 12164-c005cde/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 12164-c005cde/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 12164-c005cde/gopath/src/github.com/minio/minio fetch --quiet minio pull/12164/head:pr12164
+ git -C 12164-c005cde/gopath/src/github.com/minio/minio checkout --quiet pr12164
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/12164-c005cde/gopath
+ make -C 12164-c005cde/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading google.golang.org/api v0.25.0
go: downloading cloud.google.com/go v0.57.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/fatih/color v1.10.0
go: downloading golang.org/x/crypto v0.0.0-20210415154028-4f45737414dc
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/shirou/gopsutil/v3 v3.21.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/minio/sio v0.2.1
go: downloading github.com/tinylib/msgp v1.1.3
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
go: downloading github.com/dswarbrick/smart v0.0.0-20190505152634-909a45200d6d
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/miekg/dns v1.1.35
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/klauspost/compress v1.11.12
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/VividCortex/ewma v1.1.1
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/pierrec/lz4 v2.5.2+incompatible
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/coredns/coredns v1.4.0
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/lib/pq v1.9.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/nats-io/stan.go v0.8.3
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/willf/bitset v1.1.11
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/tidwall/gjson v1.6.8
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/tidwall/pretty v1.0.2
go: downloading github.com/tidwall/match v1.0.3
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading go.uber.org/multierr v1.5.0
go: downloading go.uber.org/atomic v1.6.0
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/nats-io/nkeys v0.2.0
go: downloading github.com/nats-io/jwt v1.1.0
go: downloading cloud.google.com/go/storage v1.8.0
go: downloading google.golang.org/protobuf v1.24.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/klauspost/cpuid/v2 v2.0.4
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading google.golang.org/grpc v1.29.1
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading google.golang.org/genproto v0.0.0-20200527145253-8367513e4ece
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: downloading github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
# github.com/minio/minio/cmd
cmd/fs-v1.go:363:26: not enough arguments in call to item.applyActions
	have (context.Context, *FSObjects, actionMeta)
	want (context.Context, ObjectLayer, actionMeta, *sizeSummary)
make: *** [build] Error 2

```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#1c0e6ee8-8879-4db5-8fc4-752a52f7bca8) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#45ae4f18-8c47-47d9-bbbb-1ca0a7a502d1) |
<a name="45ae4f18-8c47-47d9-bbbb-1ca0a7a502d1"></a>
### 12164-fd49664/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31465
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7d491778f886:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2607,
  "function": "aws --endpoint-url http://minio-c3.minio.io:31465 s3api copy-object --bucket awscli-mint-test-bucket-23651 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-23651/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 46 seconds
(7/15) Running minio-dotnet tests ... done in 43 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 30 seconds
{
  "args": {},
  "duration": 1039,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 2 minutes and 7 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 149,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-3fgnnvl, objectName = minio-java-test-158sg95-copy, resource = /minio-java-test-3fgnnvl/minio-java-test-158sg95-copy, requestId = 16799B678B951A28, hostId = 5c33c437-cb6e-4179-9014-08cd6e05d8ba)\nrequest={method=PUT, url=http://minio-c3.minio.io:31465/minio-java-test-3fgnnvl/minio-java-test-158sg95-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-2i7vfoi/minio-java-test-158sg95\nHost: minio-c3.minio.io:31465\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210427T040650Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210427/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Tue, 27 Apr 2021 04:06:50 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16799B678B951A28\nX-Xss-Protection: 1; mode=block\nDate: Tue, 27 Apr 2021 04:06:50 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 47 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 55 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2843",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-8738/s3cmd-test-object-21583 42447e48-fd29-49a8-8eb5-71824c8c8838\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 42447e48-fd29-49a8-8eb5-71824c8c8838 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
<a name="1c0e6ee8-8879-4db5-8fc4-752a52f7bca8"></a>
### 12164-fd49664/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:32080
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ccd6c8588db7:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 1 minutes and 29 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 32 seconds
(8/15) Running minio-dotnet tests ... done in 46 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 32 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 57 seconds
(11/15) Running minio-js tests ... done in 56 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 34 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-4d228ffd-4271-4dc9-9046-1a7e843d327d",
    "object_name": "58fc295c-875b-4791-a745-bd8b88ebe336"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-4d228ffd-4271-4dc9-9046-1a7e843d327d, request_id: 16799BA17D6019A2, host_id: f8836056-b12e-47c1-bc44-813a37937b5d, bucket_name: minio-py-test-4d228ffd-4271-4dc9-9046-1a7e843d327d",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-4d228ffd-4271-4dc9-9046-1a7e843d327d, request_id: 16799BA17D6019A2, host_id: f8836056-b12e-47c1-bc44-813a37937b5d, bucket_name: minio-py-test-4d228ffd-4271-4dc9-9046-1a7e843d327d\n",
  "duration": 8001
}
(12/15) Running s3cmd tests ... done in 21 seconds
(13/15) Running s3select tests ... done in 6 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/data-scanner.go [hunk 1]
```diff
@@ -936,13 +941,17 @@ func (i *scannerItem) applyLifecycle(ctx context.Context, o ObjectLayer, meta ac
...
 	// from the current deployment, which means we don't have to call healing
 	// routine even if we are asked to do via heal flag.
-	if !applied && i.heal {
-		size = i.applyHealing(ctx, o, meta)
+	if !applied {
+		if i.heal {
+			size = i.applyHealing(ctx, o, meta)
+		}
+		// replicate only if lifecycle rules are not applied.
+		i.healReplication(ctx, o, meta.oi.Clone(), sizeS)
```
Comment (poornas) [inline cmd/data-scanner.go:956]: this depends on the lifecycle action in question - if it is transition/restore action, then replication still needs to happen

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/data-scanner.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/data-scanner.go:956]: why would that happen? @poornas - have we verified such things should happen as per AWS S3?

Comment (poornas) [reply cmd/data-scanner.go:956]: Wait, you're right - I was thinking of situation when object is eligible for transition but somehow sitting in Pending/Failed replication state. Your change will only mean that scanner doesnt do both replication & transition in the same scan - that's ok, except may introduce delay in completing replication

**Review by poornas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#396bb2f2-4b89-442a-8c88-5928b4d43375) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#8f72ad83-7ed2-4df4-a038-a8b7090e7066) |
<a name="396bb2f2-4b89-442a-8c88-5928b4d43375"></a>
### 12164-d8a6411/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31421
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 39c7abe031ce:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 16 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 0 seconds
(8/15) Running minio-dotnet tests ... done in 48 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 59 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 50 seconds
(11/15) Running minio-js tests ... done in 48 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 43 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-403ce384-2a6f-417d-a65a-da5f659a8024",
    "object_name": "af61da25-fa60-40f2-b7ed-e10f05c8f949"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-403ce384-2a6f-417d-a65a-da5f659a8024, request_id: 1679A2B0E1A01772, host_id: a978d19f-fb71-400f-b79e-20f8a4919396, bucket_name: minio-py-test-403ce384-2a6f-417d-a65a-da5f659a8024",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-403ce384-2a6f-417d-a65a-da5f659a8024, request_id: 1679A2B0E1A01772, host_id: a978d19f-fb71-400f-b79e-20f8a4919396, bucket_name: minio-py-test-403ce384-2a6f-417d-a65a-da5f659a8024\n",
  "duration": 5641
}
(12/15) Running s3cmd tests ... done in 19 seconds
(13/15) Running s3select tests ... done in 6 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="8f72ad83-7ed2-4df4-a038-a8b7090e7066"></a>
### 12164-d8a6411/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30737
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp bc76813cb70e:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2957,
  "function": "aws --endpoint-url http://minio-dev7.minio.io:30737 s3api copy-object --bucket awscli-mint-test-bucket-14679 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-14679/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 42 seconds
(7/15) Running minio-dotnet tests ... done in 46 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 0 seconds
{
  "args": {},
  "duration": 527,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 2 minutes and 5 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 117,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-23rsvvg, objectName = minio-java-test-2o9c119-copy, resource = /minio-java-test-23rsvvg/minio-java-test-2o9c119-copy, requestId = 1679A266A0C1DAEB, hostId = 92aae8ee-e91b-4061-be1e-5db4c19d48c4)\nrequest={method=PUT, url=http://minio-dev7.minio.io:30737/minio-java-test-23rsvvg/minio-java-test-2o9c119-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-3jnoft8/minio-java-test-2o9c119\nHost: minio-dev7.minio.io:30737\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210427T061502Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210427/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Tue, 27 Apr 2021 06:15:02 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1679A266A0C1DAEB\nX-Xss-Protection: 1; mode=block\nDate: Tue, 27 Apr 2021 06:15:02 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 46 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 50 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2599",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-32209/s3cmd-test-object-23987 57cda74f-1fcd-4f57-920b-1513b90c4a52\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 57cda74f-1fcd-4f57-920b-1513b90c4a52 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 7 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]
LGTM

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**